src/BrowserInterop/Extensions/CallBackInteropWrapper.cs

```diff
@@ -86,15 +114,31 @@ private CallBackInteropWrapper()
         public static CallBackInteropWrapper Create(Func<ValueTask> callback, object serializationSpec = null,
             bool getJsObjectRef = false)
         {
-            var res = new CallBackInteropWrapper
+            var dotnetCallback = DotNetObjectReference.Create(new JsInteropActionWrapper(callback));
+            var res = new CallBackInteropWrapper(dotnetCallback)
             {
-                CallbackRef = DotNetObjectReference.Create(new JsInteropActionWrapper(callback)),
                 SerializationSpec = serializationSpec,
                 GetJsObjectRef = getJsObjectRef
             };
             return res;
         }
 
-        public object CallbackRef { get; set; }
+        public void Dispose()
+        {
+            //casting is safe, the constructor enforces a disposable.
+            ((IDisposable)this.CallbackRef).Dispose();
+        }
+
+
+        public object CallbackRef { get;  } // we must keep it as a object, otherwise its not serialized.
+    }
+    internal interface ICallbackReferenceData
```

Comment:
What's the point of this interface ?

Reply:
I use it to determine whether the user requests the data and the ref. see CallbackInteropWrapper line 96

Reply:
why not using the class CallbackReferenceData directly, here we leave the door open for implementation of ICallbackReferenceData without Ref/Data fields

Reply:
because CallbackReferenceData is a generic type and the type checking would be more heavy. The ICallbackReferenceData interface is internal. At least the API user cannot implement it. This concept is called marker interface.